.github/workflows/rollback.yml

```diff
@@ -0,0 +1,307 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+name: Release rollback
+
+on:
+  workflow_dispatch:
+    inputs:
+      rollback_version:
+        description: 'the version number to rollback to'
+        required: true
+
+env:
+  DOCKER_HUB_REPO: amazon/aws-otel-collector
+  ECR_REPO: aws-observability/aws-otel-collector
+  S3_BUCKET: aws-otel-collector
+  SSM_RELEASE_PACKAGE_NAME: AWSDistroOTel-Collector
+  SSM_ARTIFACT_ROOT: rollback/ssm
+
+jobs:
+  image-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check version exists on Release ECR/DockerHub repos
+        run: |
+          docker manifest inspect public.ecr.aws/$ECR_REPO:${{ github.event.inputs.rollback_version }} > /dev/null
+          docker manifest inspect $DOCKER_HUB_REPO:${{ github.event.inputs.rollback_version }} > /dev/null
+  s3-check:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      max-parallel: 5
+      matrix:
+        os: [amazon_linux, redhat, centos, suse, ubuntu, debian, windows]
+        arch: [amd64, arm64]
+        exclude:
+          - os: windows
+            arch: arm64
+        include:
+          - os: amazon_linux
+            type: rpm
+          - os: redhat
+            type: rpm
+          - os: centos
+            type: rpm
+          - os: suse
+            type: rpm
+          - os: ubuntu
+            type: deb
+          - os: debian
+            type: deb
+          - os: windows
+            type: msi
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
+          aws-region: us-west-2
+      - name: Check version exists on Release S3
+        run: |
+          S3_KEY="${{ matrix.os }}/${{ matrix.arch }}/${{ github.event.inputs.rollback_version }}/aws-otel-collector.${{ matrix.type }}"
+          aws s3api head-object --bucket $S3_BUCKET --key $S3_KEY > /dev/null
+  ssm-check:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - type: global
+            regions: "eu-north-1 ap-south-1 eu-west-3 us-east-2 eu-west-1 eu-central-1 sa-east-1 us-east-1 \
+                  ap-northeast-2 ap-northeast-3 eu-west-2 ap-northeast-1 us-west-2 us-west-1 ap-southeast-1 \
+                  ap-southeast-2 ca-central-1"
+          - type: hkg
```

Comment:
Why do we have to log in to specific region for these? I thought only gov and china required this.

Reply:
This is how the `ssm-release` workflow is set up as well. These are opt-in regions and part of different release accounts. We might not need to login for the `list-document-versions` since it's public, but I'm not sure. We definitely need to login for the `update-document-default-version`.

Reply:
Yeah just tried to run it in my branch with only the us-west-2 login and it wasn't authorized to `list-document-versions`.

Reply:
https://github.com/jefchien/aws-otel-collector/runs/4902645939?check_suite_focus=true